usr/share/rear/output/default/94_grub2_rescue.sh

```diff
@@ -110,6 +110,9 @@ else
     grub-mkconfig -o $grub_conf
 fi
 
+#Finding UUID of filesystem containing /boot 
+grub_boot_uuid=$(df /boot | awk 'END {print $1}' | xargs blkid -s UUID -o value)
```

Comment:
I agree with you - testing once is enough.
`BugIfError` function is mainly used when we think there might be a bug report needed against rear. `StopIfError` is meant to force an exit out of rear when there is a condition not met (missing executable or file or something else which is not caused by a bug)